clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5928,28 +5928,6 @@ static void handlePreferredTypeAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   S.RequireCompleteType(ParmTSI->getTypeLoc().getBeginLoc(), QT,
                         diag::err_incomplete_type);
 
-  if (QT->isEnumeralType()) {
-    auto IsCorrespondingType = [&](QualType LHS, QualType RHS) {
-      assert(LHS != RHS);
-      if (LHS->isSignedIntegerType())
-        return LHS == S.getASTContext().getCorrespondingSignedType(RHS);
-      return LHS == S.getASTContext().getCorrespondingUnsignedType(RHS);
-    };
-    QualType BitfieldType =
-        cast<FieldDecl>(D)->getType()->getCanonicalTypeUnqualified();
-    QualType EnumUnderlyingType = QT->getAs<EnumType>()
-                                      ->getDecl()
-                                      ->getIntegerType()
-                                      ->getCanonicalTypeUnqualified();
-    if (EnumUnderlyingType != BitfieldType &&
-        !IsCorrespondingType(EnumUnderlyingType, BitfieldType)) {
```

Comment:
I think we should keep the diagnostics, but only emit it when the type of the enum  has a sizeof that is greater than the type of the bitfields.

Another solution would be to compare the number of bits of the greater enumerator with the number of bits in the bitfield.

Reply:
> Another solution would be to compare the number of bits of the greater enumerator with the number of bits in the bitfield.

I've had this idea myself, but it's defeated by likes of https://github.com/llvm/llvm-project/blob/e79f0506cf6fd32ed466fdd2ccbb68dcacbb4bf5/clang/include/clang/AST/DeclarationName.h#L174-L185
where `UncommonNameKindOffset` is an enumerator that is not there to be stored in memory.

Reply:
> I think we should keep the diagnostics, but only emit it when the type of the enum has a sizeof that is greater than the type of the bitfields.

We can do that, but it's going to miss so much that I'm not sure it's worth our time at all.

Reply:
I don't really see value for keeping this one, this was not my preferred use case in the original patch.  

I Still like the 'range' one I suggested, and I don't find the `UncommonNameKindOffset` to be particularly motivating.